src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const dateObj = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateObj[fromFormat[i]] = dateParts[i];
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateObj['YY'] = dateObj['YYYY'].slice(-2);
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (dateObj.YY < 30) {
+      dateObj['YYYY'] = '20' + dateObj['YY'];
+    } else {
+      dateObj['YYYY'] = '19' + dateObj['YY'];
+    }
+  }
+
+  const newFormatDate = [];
+
+  for (let i = 0; i < 3; i++) {
```

Comment:
Similar to the previous loop, use `toFormat.length - 1` instead of '3' to avoid magic numbers and make the code more adaptable to changes in the format array.